File: interpreter/java/TeamLeadExpression.java
package Interpreter;

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
class TeamLeadExpression implements IJobExpression {
    public void interpret(Context context) {
        if (context.formula.contains("L")) {
            context.totalPoint += 7000;
        }
    }
}

File: interpreter/kotlin/TeamLeadExpression.kt
package Interpreter

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
internal class TeamLeadExpression : IJobExpression {
    override fun interpret(context: Context) {
        if (context.formula!!.contains("L")) {
            context.totalPoint = context.totalPoint + 7000
        }
    }
}
